core/src/components/radio/radio.tsx

```diff
@@ -143,6 +172,26 @@ export class Radio implements ComponentInterface {
   };
 
   render() {
+    const { legacyFormController } = this;
+
+    return legacyFormController.hasLegacyControl() ? this.renderLegacyRadio() : this.renderRadio();
+  }
+
+  private renderRadio() {
+    return <slot></slot>;
+  }
+
+  private renderLegacyRadio() {
+    if (!this.hasLoggedDeprecationWarning) {
+      printIonWarning(
+        `Using ion-radio with an ion-label has been deprecated. To migrate, remove the ion-label and pass your label directly into ion-radio instead.
+Example: <ion-radio>Email:</ion-radio>
```

Comment:
I remained consistent with toggle here, but I am questioning if showing a code snippet like:
```html
<ion-radio>Label:</ion-radio>
```
May be more intuitive/clear.

Reply:
Not sure I understand -- the code snippets look the same? (minus the label text)

Reply:
Correct, I think the default label text is a little confusing. Why would I have a radio with an `Email` label? `Option Label` would also work for me.

Reply:
Oh I see now 🤦 That's likely a copy and paste mistake from the ion-input revision.